lib/client.js

```diff
@@ -35,36 +37,32 @@ export class Http3Client extends Http3WebTransport {
     this.webtransportProm = null
     /** @type {{ resolve: (value?: any) => void, reject: (err?: Error) => void} | null | undefined} */
     this.quicconnectedProm = null
+
+    this._quicConnectTimeout = args.quicConnectTimeout ?? 8000
+    this._webTransportConnectTimeout = args.webTransportConnectTimeout ?? 2000
   }
 
   async handleConnection() {
+    this.createTransportInt()
+
     this.quicconnected = new Promise((resolve, reject) => {
       this.quicconnectedProm = { resolve, reject }
     })
     this.webtransport = new Promise((resolve, reject) => {
       this.webtransportProm = { resolve, reject }
     })
 
-    try {
-      setTimeout(() => {
-        if (this.quicconnectedProm) {
-          this.quicconnectedProm.reject(new Error('Timeout client connection'))
-          delete this.quicconnectedProm
-        }
-      }, 8000)
-      await this.quicconnected
-      // now create Webtransport session
-      setTimeout(() => {
-        if (this.webtransportProm) {
```

Comment:
Why did you drop the waiting for webtransport support. It can happen that is connects but never tell us, that it can have webtransport support.

Reply:
Because from what I can see, it doesn't wait for webtransport support, it just sets up the timer to reject the `webtransport` promise if webtransport support doesn't materialise.  The actual waiting happens in the `createWTSession` method so I moved the timer setup there so it's closer to the thing it's affecting.

From what I can see, these methods are only invoked by `lib/webtransport.js` and they are done one after another so I hope it's a bit easier to read now.

Reply:
And still, fails, if we reach the time out and no webtransport support was signaled? (The lib only has an event telling that webtransport is possible, but not if it is unsupported). Then it is fine.

Reply:
It does, but only because webtransport.js invokes one method after the other.  They could be one method, that might be a bit simpler?

Reply:
It also could be made a bit more concise by using the [pTimeout](https://github.com/fails-components/webtransport/pull/130/files#diff-277293d1d50e356ece07e2607160b2de9f4f40ff0dc859222febe7588830266c) and [defer](https://github.com/fails-components/webtransport/blob/master/lib/utils.js#L5) functions (borrowed from [p-timeout](https://www.npmjs.com/package/p-timeout) and [p-defer](https://www.npmjs.com/package/p-defer) modules) but I think better to get the tests stable before making any more changes.

Reply:
Yes one method is simpler, but it is harder to debug, what is failing on the c++ side, that is the reason it was split.

Reply:
Fair enough - hopefully some more detailed logging on the js side will help there.